File: 30 days of code( python)/day12.py
class Person:
	def __init__(self, firstName, lastName, idNumber):
		self.firstName = firstName
		self.lastName = lastName
		self.idNumber = idNumber
	def printPerson(self):
		print("Name:", self.lastName + ",", self.firstName)
		print("ID:", self.idNumber)

class Student(Person):
    #   Class Constructor
    #   
    #   Parameters:
    #   firstName - A string denoting the Person's first name.
    #   lastName - A string denoting the Person's last name.
    #   id - An integer denoting the Person's ID number.
    #   scores - An array of integers denoting the Person's test scores.
    #
    # Write your constructor here
    

    #   Function Name: calculate
    #   Return: A character denoting the grade.
    #
    # Write your function here
    def __init__(self, firstName, lastName, idNumber, testScores):
        super().__init__(firstName, lastName, idNumber)
        self.testScores = testScores

    def calculate(self):
        total = 0

        for testScore in self.testScores:
            total += testScore

        avg = total / len(self.testScores)

        if 90 <= avg <= 100:
            return 'O'
        if 80 <= avg < 90:
            return 'E'
        if 70 <= avg < 80:
            return 'A'
        if 55 <= avg < 70:
            return 'P'
        if 40 <= avg < 55:
            return 'D'
        return 'T'

line = input().split()

File: 30daysOFcode_TypeScript/day 12.ts
class Person {
    firstName: string;
    lastName: string;
    idNumber: number;

    constructor(firstName: string, lastName: string, identification: number) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.idNumber = identification;
    }

    printPerson(): void {
        console.log('Name: ' + this.lastName + ', ' + this.lastName + '\nID: ' + this.idNumber);
    }
}

class Student extends Person {
    scores: Array<number>;

    constructor(firstName: string, lastName: string, id: number, scores: Array<number>) {
        super(firstName, lastName, id);
        this.scores = scores;
    }

    calculate(): string {
        let sum: number = 0;
        for (let i: number = 0; i < this.scores.length; i++) {
            sum += this.scores[i];
        }

        let avg: number = Math.floor(sum / this.scores.length);

        if (avg >= 90 && avg <= 100) {
            return 'O';
        }
        else if (avg >= 80 && avg < 90) {
            return 'E';
        }
        else if (avg >= 70 && avg < 80) {
            return 'A';
        }
        else if (avg >= 55 && avg < 70) {
            return 'P';
        }
        else if (avg >= 40 && avg < 55) {
            return 'D';
        }
        else {
            return 'T';
        }
    }
}